### DISCUSSION TITLE
```do unsafe```, ```while (...) unsafe```, and ```else unsafe```, and more...

### DISCUSSION THREAD

**Discussion by Roshaless** [dotnet/csharplang#8493] [General] [answered]
For title, in C#, you can

``` C#
if (...) unsafe
{
    // ...
}
```

``` C#
while (...) unsafe
{
    // ...
}
```

And

``` C#
do unsafe
{
    // ...
}
while (...);
```

``` C#
if (...)
{
    // ...
}
else unsafe
{
    // ...
}
```

``` C#
lock (...) unsafe
{
    // ...
}

using (...) unsafe
{
    // ...
}
```

But this way (``unsafe``) does not work for ``checked``, ``unchecked``, ``try catch finally``.
And when you formatting it, It will have more indent (It is right? or this is a problem? idk...). 
``` C#
if (...) unsafe
    {
        // ...
    }
```

Sometimes, I don't like so many code blocks, Especially ``if else`` mix ``unsafe`` or other.
Then, I try it `` if (...) unsafe `` and found it work...👌

**Comment by huoyaoyuan** [comment, marked as answer]
It's actually brace-less blocks with nested `unsafe`. You can mix many of brace-less constructs like this:

```
if (...) using (...)
    {

    }
```

```
if (...) unchecked
   {

   }
```
and the "standard" format should be putting the inner block on new line like this:

```
if (...)
    unsafe
    {

    }
```

There is a convention to cancel the indentation if the inner and outer blocks are both `using`s. But for usual cases, every nesting level will increase the level of indentation.

**Reply by Roshaless** [reply]
Yes, the indentation level is usually increased when the inner block is different from the outer block.
But even simple statements like ``if (...) unsafe`` can't be avoided. Should this be improved? or not?

**Reply by HaloFour** [reply]
The language itself doesn't care about indentation levels.  Formatting is a question of tooling.

**Reply by Roshaless** [reply]
> The language itself doesn't care about indentation levels. Formatting is a question of tooling.

Oh, I didn't think about tooling when I create this.  But anyway, Thank you for telling me ^^

**Comment by Roshaless** [comment]
So... like this

``` C#
if (...) unsafe
{
    lock (...) using (...) try
    {
        foreach (...) fixed (...) checked
        {

        }
    }
    finally
    {
    }
}
else do unchecked
{

} while (...);
``` 

After formatting
``` C#
if (...) unsafe
    {
        lock (...) using (...) try
                {
                    foreach (...) fixed (...) checked
                            {

                            }
                }
                finally
                {
                }
    }
else do unchecked
        {

        } while (true);
```

🤔🤔🤣
